server/src/test/java/org/opensearch/index/translog/RemoteFSTranslogTests.java

```diff
@@ -570,6 +570,66 @@ public void testSimpleOperationsUpload() throws Exception {
         assertBusy(() -> assertEquals(4, translog.allUploaded().size()));
     }
 
+    public void testMetadataFileDeletion() throws Exception {
```

Comment:
can we check metadata file deletion for previous primary term as well ?

Reply:
Incorporated.